crates/cheatcodes/src/inspector.rs

```diff
@@ -1845,9 +1845,10 @@ impl Cheatcodes {
         if let Some(paused_gas) = self.gas_metering.paused_frames.last() {
             // Keep gas constant if paused.
             // Make sure we record the memory changes so that memory expansion is not paused.
-            let memory = interpreter.control.gas.memory;
+            let memory = *interpreter.control.gas.memory();
             interpreter.control.gas = *paused_gas;
-            interpreter.control.gas.memory = memory;
+            interpreter.control.gas.memory_mut().words_num = memory.words_num;
+            interpreter.control.gas.memory_mut().expansion_cost = memory.expansion_cost;
```

Comment:
cc @klkvr would you mind double checking this logic block? This should be equivalent and tests are all passing.

For context: https://github.com/bluealloy/revm/pull/2537/files